dlhub_sdk/utils/publish.py

```diff
@@ -0,0 +1,244 @@
+import logging
+import json
+import os
+import requests
+import globus_sdk
+
+from funcx.sdk.client import FuncXClient
+from funcx import ContainerSpec
+from time import sleep
+from github import Github
+import base64
+
+import mdf_toolbox
+import urllib
+
+#Uncomment if we are trying to mint identifiers
+# from identifiers_client.identifiers_api import identifiers_client, IdentifierClient
+# from identifiers_client.config import config
+
+logger = logging.getLogger(__name__)
+
+
+def create_container_spec(metadata):
+    """
+    Create the container spec for the Container Service. Iterate through
+    the dependencies and add them. Also include parsl, toolbox, home_run, etc.
+
+    :param metadata:
+    :return:
+    """
+
+    # Get the list of requirements from the schema
+    dependencies = []
+    try:
+        for k, v in metadata['dlhub']['dependencies']['python'].items():
+            dependencies.append("{0}=={1}".format(k, v))
+    except:
+        # There are no python dependencies
+        pass
+
+    model_location = None
+    if 'transfer_method' in metadata['dlhub']:
```

Comment:
what happens if `'transfer_method'` is not in the metadata blob? Can you add some clarity?

Reply:
similarly, what happens if `'S3'` or `'POST'` are not in the `'transfer_method'` blob?

Reply:
It is required that at least one of 'repository' or 'transfer_method' be defined in the metadata to be able to create the container spec.  The only code paths that call this put either the repository location or the s3 location of the container zip, but this function should explicitly raise an exception if it doesn't get one.  And, actually, transfer method can't be 'POST' anymore, so this definitely needs to be cleaned up.